src/lambda_function/s3_event_config/app.py

```diff
@@ -0,0 +1,67 @@
+"""
+This Lambda app responds to an S3 event notification and starts a Glue workflow.
+The Glue workflow name is set by the environment variable `PRIMARY_WORKFLOW_NAME`.
+Subsequently, the S3 objects which were contained in the event are written as a
+JSON string to the `messages` workflow run property.
+"""
+import os
+import json
+import logging
+import boto3
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+SUCCESS = "SUCCESS"
+FAILED = "FAILED"
+
+logger.info("Loading function")
+s3 = boto3.resource("s3")
+
+
+def lambda_handler(event, context):
+    logger.info(f"Received event: {json.dumps(event, indent=2)}")
+    try:
+        if event["RequestType"] == "Delete":
+            logger.info(f'Request Type:{event["RequestType"]}')
+            bucket = os.environ["S3_SOURCE_BUCKET_NAME"]
+            delete_notification(bucket)
+            logger.info("Sending response to custom resource after Delete")
+
+        elif event["RequestType"] == "Create" or event["RequestType"] == "Update":
+            logger.info(f'Request Type: {event["RequestType"]}')
+            lambda_arn = os.environ["S3_TO_GLUE_FUNCTION_ARN"]
+            bucket = os.environ["S3_SOURCE_BUCKET_NAME"]
+            add_notification(lambda_arn, bucket)
+            logger.info("Sending response to custom resource")
+    except Exception as e:
+        logger.info(f"Failed to process:{e}")
+
+
+def add_notification(lambda_arn: str, bucket: str) -> None:
+    """Adds the S3 notification configuration to an existing bucket
+
+    Args:
+        lambda_arn (str): Arn of the lambda s3 event config function
+        bucket (str): bucket name of the s3 bucket to add the config to
+    """
+    bucket_notification = s3.BucketNotification(bucket)
+    response = bucket_notification.put(
+        NotificationConfiguration={
+            "LambdaFunctionConfigurations": [
+                {"LambdaFunctionArn": lambda_arn, "Events": ["s3:ObjectCreated:*"]}
+            ]
+        }
+    )
+    logger.info("Put request completed....")
+
+
+def delete_notification(bucket: str) -> None:
```

Comment:
Just want to make sure that I understand that this is a lambda that adds the lambda function that will trigger jobs for S3 event notifications.  

If so, how will the delete notification be triggered?

Reply:
<img width="990" alt="image" src="https://github.com/Sage-Bionetworks/recover/assets/26471741/db2d62b8-51f5-4c12-8ac5-c4da725f8ba2">
A rough diagram of where this all falls in place. The delete notification is in place in case we'd like to delete the notification configuration/would like a more automated way of deleting the configuration. If we'd rather do that via AWS console, we can remove that code.